### PR TITLE
[review] Unify access contexts

### PR DESCRIPTION
**Pull request by dawagner**
Various reworks that suppress some static_casts and const_casts.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/278%23issuecomment-147449369%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23issuecomment-148057141%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23issuecomment-148666007%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23issuecomment-148708921%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884079%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884499%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884735%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884836%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884876%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41885727%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41902123%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41970960%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41970977%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41970986%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r42237902%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r42237903%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r42237904%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r42237906%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23issuecomment-148708958%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/278%23issuecomment-149236402%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/278%23issuecomment-147449369%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40krocard%20and%20%40tcahuzax%20Please%20review%22%2C%20%22created_at%22%3A%20%222015-10-12T16%3A14%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40krocard%20I%20should%20have%20addressed%20all%20your%20previous%20comments.%20Please%20review.%22%2C%20%22created_at%22%3A%20%222015-10-14T13%3A57%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40krocard%20please%20review%22%2C%20%22created_at%22%3A%20%222015-10-16T09%3A33%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-10-16T12%3A54%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22Need%20%40tcahuzax%20review%22%2C%20%22created_at%22%3A%20%222015-10-16T12%3A54%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40tcahuzax%20please%20review%22%2C%20%22created_at%22%3A%20%222015-10-19T14%3A44%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20c813e916d9fc7256c745c20a46230b967e7f09ef%20parameter/Element.cpp%2025%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884079%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20is%20it%20empty%20%3F%22%2C%20%22created_at%22%3A%20%222015-10-13T15%3A42%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-10-16T12%3A46%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/Element.cpp%3AL133-140%22%7D%2C%20%22Pull%20c813e916d9fc7256c745c20a46230b967e7f09ef%20utility/ErrorContext.hpp%2019%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884735%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60ErrorContext%28std%3A%3Astring%26%20error%29%20%3A%20_error%28error%29%20%7B%7D%60%22%2C%20%22created_at%22%3A%20%222015-10-13T15%3A47%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20done%22%2C%20%22created_at%22%3A%20%222015-10-14T09%3A15%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/ErrorContext.hpp%3AL32-74%22%7D%2C%20%22Pull%20c813e916d9fc7256c745c20a46230b967e7f09ef%20utility/ErrorContext.hpp%2043%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41885727%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20do%20you%20need%20so%20%3F%5Cr%5CnEDIT%3A%20I%20saw%20in%20your%20commit%20message%20d4d76accb7231d915cf4d3460f354af6c88de3de%20why%20you%20want%20such%20a%20thing.%5Cr%5Cn%20-%20could%20add%20it%20in%20the%20comment.%20The%20why%20is%20more%20important%20than%20the%20what.%20Which%20is%20fairly%20obvious%20here.%5Cr%5Cn%20-%20should%20it%20not%20be%20a%20move%20constructor%20%3F%5Cr%5Cn%20-%20The%20beaviour%20seems%20very%20implicit.%20Would%20it%20not%20be%20better%20to%20have%20a%20error%20getter%20%3F%22%2C%20%22created_at%22%3A%20%222015-10-13T15%3A54%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-10-16T12%3A46%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/ErrorContext.hpp%3AL32-74%22%7D%2C%20%22Pull%20c813e916d9fc7256c745c20a46230b967e7f09ef%20utility/ErrorContext.hpp%2053%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884876%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60merror%60%22%2C%20%22created_at%22%3A%20%222015-10-13T15%3A48%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20done%22%2C%20%22created_at%22%3A%20%222015-10-14T09%3A16%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/ErrorContext.hpp%3AL32-74%22%7D%2C%20%22Pull%20c813e916d9fc7256c745c20a46230b967e7f09ef%20parameter/ParameterAccessContext.h%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884499%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%7E%7E%7Ecpp%5Cr%5CnCParameterAccessContext%28const%20utility%3A%3AErrorContext%20%26other%29%5Cr%5Cn%20%20%20%20%20%3A%20utility%3A%3AErrorContext%28other%29%20%7B%7D%5Cr%5Cn%7E%7E%7E%22%2C%20%22created_at%22%3A%20%222015-10-13T15%3A45%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-10-16T12%3A46%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/ParameterAccessContext.h%3AL79-91%22%7D%2C%20%22Pull%20c813e916d9fc7256c745c20a46230b967e7f09ef%20utility/ErrorContext.hpp%2046%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41884836%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20are%20you%20calling%20the%20NonCopyable%20class%20%3F%3F%3F%22%2C%20%22created_at%22%3A%20%222015-10-13T15%3A48%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20done%22%2C%20%22created_at%22%3A%20%222015-10-14T09%3A16%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/ErrorContext.hpp%3AL32-74%22%7D%2C%20%22Pull%20c813e916d9fc7256c745c20a46230b967e7f09ef%20parameter/DomainConfiguration.cpp%20149%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/278%23discussion_r41902123%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Should%20this%20rawness%20not%20be%20part%20of%20a%20common%20base%20class%20between%20accessContext%20and%20export%20context...%20Anyway%20this%20is%20fixed%20by%20Francois%27%20rework%20to%20make%20configurationAccessContext%20part%20of%20Serializing%20context.%20Will%20your%20rework%20invalidate%20Francois%27%20%3F%22%2C%20%22created_at%22%3A%20%222015-10-13T18%3A04%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-10-16T12%3A46%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/DomainConfiguration.cpp%3AL118-192%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/krocard'></a>

- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/278#issuecomment-147449369'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> @krocard and @tcahuzax Please review
- <a href='https://github.com/dawagner'></a> @krocard I should have addressed all your previous comments. Please review.
- <a href='https://github.com/dawagner'></a> @krocard please review
- <a href='https://github.com/krocard'></a> :shipit:
- <a href='https://github.com/krocard'></a> Need @tcahuzax review
- <a href='https://github.com/dawagner'></a> @tcahuzax please review
- [x] <a href='#crh-comment-Pull c813e916d9fc7256c745c20a46230b967e7f09ef parameter/Element.cpp 25'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/278#discussion_r41884079'>File: parameter/Element.cpp:L133-140</a></b>
- <a href='https://github.com/krocard'></a> why is it empty ?
- [x] <a href='#crh-comment-Pull c813e916d9fc7256c745c20a46230b967e7f09ef parameter/ParameterAccessContext.h 21'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/278#discussion_r41884499'>File: parameter/ParameterAccessContext.h:L79-91</a></b>
- <a href='https://github.com/krocard'></a>
~~~cpp
CParameterAccessContext(const utility::ErrorContext &other)
: utility::ErrorContext(other) {}
~~~
- [x] <a href='#crh-comment-Pull c813e916d9fc7256c745c20a46230b967e7f09ef utility/ErrorContext.hpp 19'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/278#discussion_r41884735'>File: utility/ErrorContext.hpp:L32-74</a></b>
- <a href='https://github.com/krocard'></a> `ErrorContext(std::string& error) : _error(error) {}`
- <a href='https://github.com/dawagner'></a> :+1: done
- [x] <a href='#crh-comment-Pull c813e916d9fc7256c745c20a46230b967e7f09ef utility/ErrorContext.hpp 46'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/278#discussion_r41884836'>File: utility/ErrorContext.hpp:L32-74</a></b>
- <a href='https://github.com/krocard'></a> why are you calling the NonCopyable class ???
- <a href='https://github.com/dawagner'></a> :+1: done
- [x] <a href='#crh-comment-Pull c813e916d9fc7256c745c20a46230b967e7f09ef utility/ErrorContext.hpp 53'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/278#discussion_r41884876'>File: utility/ErrorContext.hpp:L32-74</a></b>
- <a href='https://github.com/krocard'></a> `merror`
- <a href='https://github.com/dawagner'></a> :+1: done
- [x] <a href='#crh-comment-Pull c813e916d9fc7256c745c20a46230b967e7f09ef utility/ErrorContext.hpp 43'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/278#discussion_r41885727'>File: utility/ErrorContext.hpp:L32-74</a></b>
- <a href='https://github.com/krocard'></a> why do you need so ?
EDIT: I saw in your commit message d4d76accb7231d915cf4d3460f354af6c88de3de why you want such a thing.
- could add it in the comment. The why is more important than the what. Which is fairly obvious here.
- should it not be a move constructor ?
- The beaviour seems very implicit. Would it not be better to have a error getter ?
- [x] <a href='#crh-comment-Pull c813e916d9fc7256c745c20a46230b967e7f09ef parameter/DomainConfiguration.cpp 149'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/278#discussion_r41902123'>File: parameter/DomainConfiguration.cpp:L118-192</a></b>
- <a href='https://github.com/krocard'></a> Should this rawness not be part of a common base class between accessContext and export context... Anyway this is fixed by Francois' rework to make configurationAccessContext part of Serializing context. Will your rework invalidate Francois' ?

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/278?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/278?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/278?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/278'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
@krocard and @tcahuzax Please review

File: parameter/Element.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ void CElement::showDescriptionProperty(std::string &strResult) const
 }
 
 // Content dumping
-void CElement::logValue(string& /*strValue*/, CErrorContext& /*ctx*/) const
+void CElement::logValue(string& /*strValue*/, utility::ErrorContext& /*ctx*/) const
 {
 }
```
Comment (krocard) [inline parameter/Element.cpp]: why is it empty ?

File: parameter/ParameterAccessContext.h [hunk 2]
```diff
@@ -79,6 +79,12 @@ class CParameterAccessContext : public CErrorContext
     void setBaseOffset(size_t baseOffset);
     size_t getBaseOffset() const;
 
+protected:
+    CParameterAccessContext(const utility::ErrorContext &other)
+        : utility::ErrorContext(other)
```
Comment (krocard) [inline parameter/ParameterAccessContext.h]: CParameterAccessContext(const utility::ErrorContext &other)
     : utility::ErrorContext(other) {}

File: utility/ErrorContext.hpp [hunk 3]
```diff
@@ -32,18 +32,42 @@
...
 
-class CErrorContext : private utility::NonCopyable
+namespace utility
+{
+
+class ErrorContext : private NonCopyable
 {
 public:
-    CErrorContext(std::string& strError);
+    ErrorContext(std::string& error) : _error(error)
```
Comment (krocard) [inline utility/ErrorContext.hpp]: `ErrorContext(std::string& error) : _error(error) {}`

File: utility/ErrorContext.hpp [hunk 4]
```diff
@@ -32,18 +32,42 @@
...
 
-    // Error
-    void setError(const std::string& strError);
-    void appendToError(const std::string& strAppend);
-    const std::string& getError() const;
+protected:
+    /** Make ErrorContext copyable but only for sub-classes
+     */
+    ErrorContext(const ErrorContext &other)
+        : NonCopyable(), _error(other._error)
```
Comment (krocard) [inline utility/ErrorContext.hpp]: why are you calling the NonCopyable class ???

File: utility/ErrorContext.hpp [hunk 5]
```diff
@@ -32,18 +32,42 @@
...
+    {
+    }
 
 private:
     // Error reference
-    std::string& _strError;
+    std::string& _error;
```
Comment (krocard) [inline utility/ErrorContext.hpp]: `merror`

File: utility/ErrorContext.hpp [hunk 6]
```diff
@@ -32,18 +32,42 @@
...
+    {
+        _error = prepend + _error;
+    }
 
-    // Error
-    void setError(const std::string& strError);
-    void appendToError(const std::string& strAppend);
-    const std::string& getError() const;
+protected:
+    /** Make ErrorContext copyable but only for sub-classes
```
Comment (krocard) [inline utility/ErrorContext.hpp]: why do you need so ?

File: parameter/DomainConfiguration.cpp [hunk 7]
```diff
@@ -121,72 +118,74 @@ void CDomainConfiguration::composeSettings(CXmlElement& xmlConfigurationSettings
...
+    CXmlElement xmlConfigurableElementSettingsElementContent;
+    xmlConfigurableElementSettingsElement.createChild(xmlConfigurableElementSettingsElementContent, source->getKind());
+    xmlConfigurableElementSettingsElementContent.setNameAttribute(source->getName());
 
-        // Forward error
-        xmlDomainExportContext.setError(strError);
+    // Create configuration access context
+    CConfigurationAccessContext configurationAccessContext(context, true);
+    configurationAccessContext.setValueSpaceRaw(context.valueSpaceIsRaw());
+    configurationAccessContext.setOutputRawFormat(context.outputRawFormatIsHex());
```
Comment (krocard) [inline parameter/DomainConfiguration.cpp]: Should this rawness not be part of a common base class between accessContext and export context... Anyway this is fixed by Francois's rework to make configurationAccessContext part of Serializing context. Will your rework invalidate Francois's ?

File: utility/ErrorContext.hpp (hunk 3 above)
Comment (dawagner) [inline utility/ErrorContext.hpp]: :+1: done

File: utility/ErrorContext.hpp (hunk 4 above)
Comment (dawagner) [inline utility/ErrorContext.hpp]: :+1: done

File: utility/ErrorContext.hpp (hunk 5 above)
Comment (dawagner) [inline utility/ErrorContext.hpp]: :+1: done

**Comment by dawagner** [comment]
@krocard I should have addressed all your previous comments. Please review.

**Comment by dawagner** [comment]
@krocard please review

File: parameter/Element.cpp (hunk 1 above)
Comment (krocard) [inline parameter/Element.cpp]: :+1:<a href='#crh-update-none'></a>

File: utility/ErrorContext.hpp (hunk 6 above)
Comment (krocard) [inline utility/ErrorContext.hpp]: :+1:<a href='#crh-update-none'></a>

File: parameter/ParameterAccessContext.h (hunk 2 above)
Comment (krocard) [inline parameter/ParameterAccessContext.h]: :+1:<a href='#crh-update-none'></a>

File: parameter/DomainConfiguration.cpp (hunk 7 above)
Comment (krocard) [inline parameter/DomainConfiguration.cpp]: :+1:<a href='#crh-update-complete'></a>

**Comment by krocard** [comment]
:shipit:

**Comment by krocard** [comment]
Need @tcahuzax review

**Comment by dawagner** [comment]
@tcahuzax please review

**Pull request merged**